### PR TITLE
Add proxy vars to configure OCP cluster wide proxy

### PR DESCRIPTION
**Pull request by stuggi**

### CODE REVIEW

File: ansible/vars/default.yaml [hunk 1]
```diff
@@ -45,6 +45,11 @@ ocp_release_image: "quay.io/openshift-release-dev/ocp-release:{{ ocp_version }}.
 ocp_release_data_url: "https://amd64.ocp.releases.ci.openshift.org/releasestream/4-stable/release/{{ ocp_version }}.{{ ocp_minor_version }}"
 ocp_release_type: ga
 
+# OCP cluster wide proxy settings `oc get proxy/cluster`
+ocp_http_proxy: "http://squid.corp.redhat.com:3128"
+ocp_https_proxy: "http://squid.corp.redhat.com:3128"
+ocp_no_proxy: ".apps.ostest.{{ base_domain_name }},.usersys.redhat.com,192.168.111.0/24"
```
Comment (stuggi) [inline ansible/vars/default.yaml:51]: not sure if we should set this in the job dev instead of here, thoughts?

Comment (olliewalsh) [reply ansible/vars/default.yaml:51]: maybe leave the conf here but add a flag to enable it (disabled by default) that we can flip for individual jobs

File: ansible/templates/ai/crucible/inventory.ospd.yml.j2 [hunk 2]
```diff
@@ -38,6 +38,20 @@ all:
     # network_type: OpenShiftSDN
     network_type: {{ ocp_network_type }}
 
+    # Proxy settings. These settings apply to: Assisted Installer, Day1 clusters and Day2 clusters.
+    # This assumes the host where the AI runs and the OpenShift cluster share the same proxy settings.
+{% if enable_ocp_cluster_wide_proxy is defined and enable_ocp_cluster_wide_proxy is sameas true %}
```
Comment (olliewalsh) [inline ansible/templates/ai/crucible/inventory.ospd.yml.j2:43]: Never seen `is sameas true`. I assume that's equivalent to `var|bool == true`

Comment (olliewalsh) [reply ansible/templates/ai/crucible/inventory.ospd.yml.j2:43]: Could also have `var|default(false) is sameas true` to avoid the `is defined` check

**Review by olliewalsh — APPROVED** [review]
/lgtm

**Pull request merged**